tracer/src/Datadog.Trace.Manual/SpanContextExtractor.cs

```diff
@@ -43,13 +43,13 @@ public SpanContextExtractor()
         /// multiple times may lead to incorrect stats when using Data Streams Monitoring.
         /// </summary>
         /// <param name="carrier">The carrier of the SpanContext. Often a header (http, kafka message header...)</param>
-        /// <param name="getter">Given a key name, returns values from the carrier. Should return an empty collection if the requested key is not present.</param>
+        /// <param name="getter">Given a key name, returns values from the carrier. Should return an empty array if the requested key is not present.</param>
         /// <param name="messageType">For Data Streams Monitoring: The type of messaging system where the message is coming from.</param>
         /// <param name="source">For Data Streams Monitoring: The queue or topic where the message is coming from.</param>
         /// <typeparam name="TCarrier">Type of the carrier</typeparam>
         /// <returns>A potentially null Datadog SpanContext</returns>
         [Instrumented]
-        public ISpanContext? ExtractIncludingDsm<TCarrier>(TCarrier carrier, Func<TCarrier, string, IEnumerable<string?>> getter, string messageType, string source)
+        public ISpanContext? ExtractIncludingDsm<TCarrier>(TCarrier carrier, Func<TCarrier, string, string?[]> getter, string messageType, string source)
```

Comment:
Hmmm, I'm not a fan of this change, as it makes the carrier function different to the other APIs 🤔 I understand why you did it, but I just don't think it's worth it 🤔